idaes/core/control_volume0d.py

```diff
@@ -1531,112 +1531,243 @@ def _get_stream_table_contents(self, time_point=0):
+                if iscale.get_scaling_factor(v) is None:
+                    sf = iscale.get_scaling_factor(
+                        self.properties_out[t].get_material_flow_terms(p, j),
+                        default=1, warning=True)
+                    iscale.set_scaling_factor(v, sf)
+
+        if hasattr(self, "phase_equilibrium_generation"):
+            for (t, e), v in self.phase_equilibrium_generation.items():
+                if iscale.get_scaling_factor(v) is None:
+                    sf = iscale.min_scaling_factor(
+                        [self.properties_out[t].get_material_flow_terms(p, j)
+                         for p in phase_list])
+                    iscale.set_scaling_factor(v, sf)
+
+        if hasattr(self, "mass_transfer_term"):
+            for (t, p, j), v in self.mass_transfer_term.items():
+                if iscale.get_scaling_factor(v) is None:
+                    sf = iscale.get_scaling_factor(
+                        self.properties_out[t].get_material_flow_terms(p, j),
+                        default=1, warning=True)
+                    iscale.set_scaling_factor(v, sf)
+
+        if hasattr(self, "rate_reaction_extent"):
+            if iscale.get_scaling_factor(self.rate_reaction_extent) is None:
+                # extent of reaction typically between 0.01 and 1
+                iscale.set_scaling_factor(self.rate_reaction_extent, 10)
+
+        if hasattr(self, "equilibrium_reaction_extent"):
+            if iscale.get_scaling_factor(self.equilibrium_reaction_extent) is None:
+                # extent of reaction typically between 0.01 and 1
+                iscale.set_scaling_factor(self.equilibrium_reaction_extent, 10)
+
+        if hasattr(self, "element_holdup"):
+            for (t, e), v in self.element_holdup.items():
+                # minimum scaling factor for elemental_hold up terms
+                sf_list = []
+                flow_basis = self.properties_out[t].get_material_flow_basis()
+                for p, j in phase_component_set:
+                    sf = iscale.get_scaling_factor(self.volume[t])
+                    if flow_basis == MaterialFlowBasis.molar:
+                        sf *= 1
+                    elif flow_basis == MaterialFlowBasis.mass:
+                        # MW scaling factor is the inverse of its value
+                        sf *= 1/(self.properties_out[t].mw_comp[j])**-1
+                    sf *= iscale.get_scaling_factor(
+                        self.phase_fraction[t, p],
+                        default=10, warning=True)
+                    sf *= iscale.get_scaling_factor(
+                        self.properties_out[t].get_material_density_terms(p, j),
+                        default=1, warning=True)
+                    # element comp scaling factor is the inverse of its value
+                    sf *= (self.properties_out[t].params.element_comp[j][e])**-1
+                    sf_list.append(sf)
+                sf = min(sf_list)
```

Comment:
1. We should set separate scaling factors for each component where possible, as components might have very different scaling.
2. Elemental holdup scaling should use the same approach as the normal material scaling terms, just with the addition of the stoichiometric conversion (and conversion from mass basis if appropriate).